src/cascadia/TerminalApp/GlobalAppSettings.cpp

```diff
@@ -280,3 +319,115 @@ std::wstring_view GlobalAppSettings::_SerializeTheme(const ElementTheme theme) n
         return SystemThemeValue;
     }
 }
+
+// Method Description:
+// - Helper function for converting the initial position string into
+//   2 coordinate values. We allow users to only provide one coordinate,
+//   thus, we use comma as the separater:
+//   (100, 100): standard input string
+//   (, 100), (100, ): if a value is missing, we set this value as a default
+//   (,): both x and y are set to default
+//   (abc, 100): if a value is not valid, we treat it as default
+//   (100, 100, 100): we only read the first two values, this is equivalent to (100, 100)
+// Arguments:
+// - initialPosition: the initial position string from json
+//   initialX: reference to the _initialX member
+//   initialY: reference to the _initialY member
+// Return Value:
+// - None
+void GlobalAppSettings::_ParseInitialPosition(const std::wstring& initialPosition,
+                                              std::optional<int32_t>& initialX,
+                                              std::optional<int32_t>& initialY) noexcept
+{
+    const wchar_t singleCharDelim = L',';
+    std::wstringstream tokenStream(initialPosition.c_str());
+    std::wstring token;
+    uint8_t initialPosIndex = 0;
+    size_t* idx = nullptr;
+
+    // Get initial position values till we run out of delimiter separated values in the stream
+    // or we hit max number of allowable values (= 2)
+    // Non-numeral values or empty string will be caught as exception and we do not assign them
+    for (; std::getline(tokenStream, token, singleCharDelim) && (initialPosIndex < 2); initialPosIndex++)
+    {
+        try
+        {
+            int32_t position = std::stoi(token, idx);
+            if (initialPosIndex == 0)
+            {
+                initialX.emplace(position);
+            }
+
+            if (initialPosIndex == 1)
+            {
+                initialY.emplace(position);
+            }
+        }
+        catch (...)
+        {
+            // Do nothing
+        }
+    }
+}
+
+// Method Description:
+// - Helper function for converting X/Y initial positions to a string
+//   value.
+// Arguments:
+// - initialX: reference to the _initialX member
+//   initialY: reference to the _initialY member
+// Return Value:
+// - The concatenated string for the the current initialX and initialY
+std::string GlobalAppSettings::_SerializeInitialPosition(const std::optional<int32_t>& initialX,
+                                                         const std::optional<int32_t>& initialY) noexcept
+{
+    std::string serializedInitialPos = "(";

```

Comment:
the deserializer doesn't use `(` or `)`, should we just use `100, 200` instead? I don't think the user should have to type `(100, 200)` :smile:

Reply:
removed, thanks for this reminding !